sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -240,4 +250,14 @@ private static Value getValue(Resource resource) {
     public String toString() {
         return ToStringBuilder.reflectionToString(this);
     }
+
+    public ResourceBuilder setRole(Optional<String> role) {
```

Comment:
i liked the prior location next to the other setters

is this meant to support clearing role/principal via `set[Role|Principal](Optional.empty())`?